flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java

```diff
@@ -189,6 +215,9 @@ public FlinkKinesisConsumer(List<String> streams, KinesisDeserializationSchema<T
 		// check the configuration properties for any conflicting settings
 		KinesisConfigUtil.validateConsumerConfiguration(this.configProps);
 
+		checkNotNull(awsClientConfig, "awsClientConfig can not be null");
```

Comment:
Since it is not required today, it should probably be optional. Then, there is no need to change the constructor but add a separate setter, like it was done for shardAssigner.

Reply:
Makes sense